ansible/server-state-playbooks/nightshade-web/playbook.yml

```diff
@@ -1,35 +1,16 @@
 # Install OMERO.web with a public user on localhost
+# Pass the command line 'upgrade-webapps' flag to upgrade webapps
 
 - hosts: all
 
-  pre_tasks: 
+  pre_tasks:
 
-    # For OMERO.web apps
-    # Pythonpath must exist before omero.web.config is provisioned
-    # or OMERO.web won't start.
-    # create systemd file addition for PYTHONPATH
-    - name: OMERO.web apps | (pre_task) configure systemd for pythonpath - create config folder 
-      become: yes
-      file:
-        dest: "/etc/systemd/system/omero-web.service.d"
-        state: directory
-        mode: "u=rwx,go=rx" 
-        owner: "root" 
-        group: "root" 
-
-    # systemd web-apps folder to pythonpath to env
-    - name: OMERO.web apps | (pre_task) configure systemd for pythonpath 
-      become: yes
-      blockinfile:
-        create: yes
-        destfile: /etc/systemd/system/omero-web.service.d/pythonpath.conf
-        owner: "root" 
-        group: "root" 
-        block: |2+
-          [Service]
-          Environment="PYTHONPATH=$PYTHONPATH:/opt/omero/web/web-extensions"
-      notify:
-        - reload systemd
+    # Set desired state of webapps (upgrade v.s. install/keep current version)
+    - set_fact:
+        webapps_state: present 
```

Comment:
Yep - thanks, I couldn't find how to do make that conditional just work, without registering the fact. I read that indeed it would persist, so made sure to initialise it first with no conditional to `present`. Thanks for the syntax, and I'll refactor this as you suggest.